src/components/footer.js

```diff
@@ -67,21 +67,77 @@ const CustomWrap = styled.div`
   text-align: center;
 `;
 
-const Footer = () => (
-  <FooterWrap>
-    <div className={wrap} />
-    <CustomWrap />
-    <div className={wrap}>
-      <Dropdown>
-        <div className={dropdownContent}>
-          <Element>1</Element>
-          <Element>2</Element>
-          <Element>3</Element>
+class Footer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentLanguage: 'en',
+    };
+  }
+
+  changeLanguage = (lng) => {
+    i18n.changeLanguage(lng);
```

Comment:
changelanguage zwraca promise więc lepiej by było ustawić state na callbacku changeLanguage żeby być pewnym, że język się w ogóle zmieni, nie ma błędów i dzieje się to PO zmianie. Tu przykładz docsów:
`i18next
  .changeLanguage('en')
  .then((t) => {
    t('key'); // -> same as i18next.t
  });`

Więcej tu: https://www.i18next.com/overview/api#changelanguage